src/Workspaces/Core/Portable/Workspace/Solution/DocumentInfo.cs

```diff
@@ -94,6 +94,22 @@ internal DocumentInfo(DocumentAttributes attributes, TextLoader? loader, IDocume
                 documentServiceProvider: null);
         }
 
+        // TODO: https://github.com/dotnet/roslyn/issues/35079
+        // Used by Razor: https://github.com/dotnet/aspnetcore-tooling/blob/master/src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioMacDocumentInfoFactory.cs#L38
+        [Obsolete]
```

Comment:
```suggestion
        [Obsolete("This is a compatibility shim for Razor; please do not use it.")]
```

We do this for a bunch of IVTs in other places; this makes it easy to grep for things that can be deleted and means we don't get more debt in the mean time.